static/app/components/notificationActions/forms/pagerdutyForm.tsx

```diff
@@ -0,0 +1,141 @@
+import {useMemo, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import DropdownButton from 'sentry/components/dropdownButton';
+import {DropdownMenu, MenuItemProps} from 'sentry/components/dropdownMenu';
+// import {
+//   NotificationActionCell,
+//   NotificationActionFormContainer,
+// } from 'sentry/components/notificationActions/notificationActionItem';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {
+  AvailableNotificationAction,
+  NotificationAction,
+} from 'sentry/types/notificationActions';
+
+type PagerdutyFormProps = {
+  /**
+   * The notification action being represented
+   */
+  action: Partial<NotificationAction>;
+  onCancel: () => void;
+  onChange: (name: string, value: any) => void;
+  onSave: () => void;
+  /**
+   * Map of pagerduty integration IDs to available actions for those IDs
+   */
+  pagerdutyIntegrations: Record<number, AvailableNotificationAction[]>;
+};
+
+const PagerdutyForm = ({
+  action,
+  onCancel,
+  onChange,
+  onSave,
+  pagerdutyIntegrations,
+}: PagerdutyFormProps) => {
+  const [selectedAccount, setSelectedAccount] = useState(
+    action.integrationId
+      ? pagerdutyIntegrations[action.integrationId][0].action.integrationName
+      : ''
+  );
+  const [selectedDisplay, setSelectedDisplay] = useState(action.targetDisplay ?? '');
+
+  const accountOptions = useMemo(() => {
+    return Object.keys(pagerdutyIntegrations).map<MenuItemProps>(integrationId => {
+      // Get the name of the integration for the integrationId from the first
+      // AvailableNotificationAction element in the array
+      const integrationName =
+        pagerdutyIntegrations[integrationId][0].action.integrationName;
+      return {
+        key: integrationName,
+        label: integrationName,
+        onAction: value => {
+          onChange('integrationId', integrationId);
+          setSelectedAccount(value);
+        },
+      };
+    });
+  }, [pagerdutyIntegrations, onChange]);
+
+  // Each Pagerduty account has its own list of services
+  const getServiceOptions = (): MenuItemProps[] => {
```

Comment:
🤔 hmmm
we can toss the type on the const, rather than the function right?
```
const getServiceOptions: MenuItemProps[] = () => {}
```

Reply:
typescript does not like that 🤷‍♀️